internal/dialect/dialectquery/clickhouse.go

```diff
@@ -34,6 +59,20 @@ func (c *Clickhouse) GetMigrationByVersion(tableName string) string {
 }
 
 func (c *Clickhouse) ListMigrations(tableName string) string {
-	q := `SELECT version_id, is_applied FROM %s ORDER BY version_id DESC`
+	q := `SELECT version_id, is_applied FROM %s ORDER BY tstamp DESC`
```

Comment:
I believe this is fixing a bug? (I think you mentioned this in the description)

The core goose library relies on a non-user-defined order for when migrations get applied, this is typically a sequential id in all other databases. It seems here we were using `version_id`, but this is incorrect. In lieu of a sequential id, we should be using time..